src/v/net/connection_rate_counter.h

```diff
@@ -80,6 +85,8 @@ class connection_rate_counter {
     std::chrono::milliseconds one_token_time;
 
 private:
+    bool need_wait() { return avaiable_new_connections() == 0; }
```

Comment:
is this counting something different than the semaphore? the semaphore provides try_wait interface which seems similar to `need_wait()`?